### ISSUE TITLE
Why do you divide the rtn[1] by 100 ?

### ISSUE THREAD

**Issue by CoinCheung** [ZhenyueQin/Research-Softmax-with-Mutual-Information#1]
https://github.com/ZhenyueQin/Research-Softmax-with-Mutual-Information/blob/1894b64709f8612d463ef3198efec3805da91374/pc_softmax_higher_accuracy/mnist_experiments/classification_mnist.py#L33

Hi, 

Thanks for releasing the code !!

 I noticed that the equatin (10) in the paper gives the definition of pc-softmax that the coefficients of the logits exps are the distributions of the labels. In the code, the distributions are computed as the proportion of the labels in the dataset. But I do not understand why you divide the `rtn[1]` by 100 at this line. Would you please tell me what is the difference between this rtn[1] and other elements in the python list, and why do we need to divide it by 100?

**Comment by kfzyqin** [comment]
(Please ignore this answer. This is wrong, sorry)

Hi, 

Thanks for reading the paper and code. 

The division of 100 is to get the percentage (%) of each label. It is used to help to know the proportions of the instances within each label. In fact, that line is only for debugging purposes I assume.

**Comment by CoinCheung** [comment]
In this line, the label number has already been divided by the total number:  
https://github.com/ZhenyueQin/Research-Softmax-with-Mutual-Information/blob/1894b64709f8612d463ef3198efec3805da91374/pc_softmax_higher_accuracy/mnist_experiments/classification_mnist.py#L29

That means the elements in the list is already within the range of (0, 1). If the percentage is needed, shouldn't we multiply the elements by 100 rather divided by 100? By the way, why do we only divide `rtn[1]` by 100, but leave other elements in the list of `rtn` unchanged?

**Comment by kfzyqin** [comment]
Hi Coin, 

Thank you for your careful picking up. You are right, that line is not for obtaining the percentage. Please just ignore that line, I have made a new commit and deleted that line:
`rtn[1]`
. That line was not in the implementation of the paper. 

Purpose of `rtn[1]`: 
I wish to understand why PC-softmax produces higher accuracy than traditional softmax after the paper has been produced. Intuitively, when the dataset is unbalanced, PC-softmax will assign higher weights to the labels that have fewer instances. The purpose of that line is to investigate whether assigning a higher weight to label 1 can improve the accuracy of classifying data corresponding to label 1.

**Comment by CoinCheung** [comment]
Thanks for explaining !!!
